### PR TITLE
feat: [SRTP-1978] add v2 support for RTP send API and update related tests

### PR DESCRIPTION
**Pull request by michele-fucito**
### Description

This PR adds a new set of functional tests for the RTP send flow using the Web API v2 endpoint.
The v2 endpoint is identical to v1 (`POST /rtps`) but uses the `Version: v2` HTTP header.
All existing `_THROUGH_WEB_API` test cases have been duplicated as `_THROUGH_WEB_API_V2` variants
across all send test files (MOCK, CBI, ICCREA, POSTE).
A dedicated set of mock fiscal codes has been introduced for v2 scenarios to avoid interference
with the v1 test data.

### List of changes

- Added `send_rtp_v2` function in `api/RTP_send_api.py` (same as `send_rtp` but with `Version: v2` header)
- Added `send_rtp_v2_and_get_status_via_rest` and `send_rtp_v2_and_get_status_by_notice_number_via_rest` helpers in `utils/rtp_send_helpers.py`
- Added `_THROUGH_WEB_API_V2` test variants in:
  - `functional-tests/tests/send_rtp/test_RTP_send_MOCK.py` (8 tests)
  - `functional-tests/tests/send_rtp/test_RTP_send_CBI.py` (1 test)
  - `functional-tests/tests/send_rtp/test_RTP_send_ICCREA.py` (1 test)
  - `functional-tests/tests/send_rtp/test_RTP_send_POSTE.py` (1 test)
- Added 7 new v2 mock fiscal code variables (`MOCK_*_V2`) in `.env`, `.env.example`, `config/configuration.py` and `.github/workflows/run_tests.yml`

### Motivation and context

The platform is introducing a v2 version of the RTP send Web API. These tests ensure full parity
of coverage between v1 and v2, covering happy paths, rejection scenarios, and edge cases
(missing `_links`, extra fields in EPC response).

### Aggregation level

- [ ] Test case
- [x] Test suite

### Test type

- [x] Functional test
- [ ] Bdd test
- [ ] Performance
- [ ] End to end
- [ ] Performance

### Type of changes

- [x] Add new test case/suite
- [ ] Modify existing test case/suite

### Test Results

- [ ] Success
- [x] Failure
- [ ] Poor performance
- [ ] Good performance

> ⚠️ `test_send_rtp_sync_sent_extra_field_THROUGH_WEB_API_V2` is currently **failing**: the RTP status
> transitions to `ACCEPTED` instead of `SENT` when the EPC v2 response contains an extra field.
> This is a known bug on the platform side (v1 equivalent passes correctly).

### Did you update secrets accordingly?

- [x] Yes
  - [ ] `.secrets_template.yaml`
  - [ ] Pipelines' secure file
  - [ ] Confluence
- [ ] Not needed
- [ ] No (_please provide further information_)

> 7 new GitHub Repository Variables must be added: `MOCK_ACTC_FISCAL_CODE_V2`, `MOCK_RJCT_FISCAL_CODE_V2`,
> `MOCK_NO_LINKS_FISCAL_CODE_V2`, `MOCK_EXTRA_FIELD_FISCAL_CODE_V2`, `MOCK_SERVER_ERROR_FISCAL_CODE_V2`,
> `MOCK_RJCT_EXTRA_FIELD_FISCAL_CODE_V2`, `MOCK_RJCT_NO_LINKS_FISCAL_CODE_V2`.

### Did you update dependencies accordingly?

- [ ] Yes (which requirements.txt did you update?)
- [x] Not needed

### CODE REVIEW

File: functional-tests/tests/send_rtp/test_RTP_send_MOCK.py [hunk 1]
```diff
@@ -392,3 +394,182 @@ def test_send_rtp_sync_rejected_no_links_THROUGH_WEB_API(
...
+@allure.story(
+    "Service provider sends an RTP with a non-compliant synchronous ACTC-like response containing an unexpected field that is ignored"
+)
+@allure.title(
+    "An RTP sent when EPC response contains an unknown field is ignored and remains in status SENT - through Web API V2"
+)
+@allure.tag("functional", "unhappy_path", "rtp_send", "optional_epc_fields")
+@pytest.mark.send
+@pytest.mark.unhappy_path
+def test_send_rtp_sync_sent_extra_field_THROUGH_WEB_API_V2(
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply functional-tests/tests/send_rtp/test_RTP_send_MOCK.py:518]: we accept to have a failure for the meantine.

File: api/RTP_send_api.py [hunk 2]
```diff
@@ -17,5 +17,18 @@ def send_rtp(access_token: str, rtp_payload):
     )
 
 
+def send_rtp_v2(access_token: str, rtp_payload):
+    """API to post a rtp request using Version v2 header.
+    :returns: the response of the call.
+    :rtype: requests.Response
+    """
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply api/RTP_send_api.py]: solved

File: api/RTP_send_api.py [hunk 3]
```diff
@@ -4,18 +4,30 @@
 from api.utils.http_utils import HTTP_TIMEOUT
 
 
-def send_rtp(access_token: str, rtp_payload):
-    """API to post a rtp request
-    :returns: the response of the call.
-    :rtype: requests.Response
-    """
+def _post_rtp(access_token: str, rtp_payload, version: str) -> requests.Response:
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply api/RTP_send_api.py]: solved

File: api/RTP_send_api.py [hunk 4]
```diff
@@ -4,18 +4,30 @@
...
+    """
+    return _post_rtp(access_token, rtp_payload, "v1")
+
+
+def send_rtp_v2(access_token: str, rtp_payload):
+    """API to post a rtp request (Version: v2).
+    :returns: the response of the call.
+    :rtype: requests.Response
+    """
+    return _post_rtp(access_token, rtp_payload, "v2")
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply api/RTP_send_api.py]: solved

File: utils/rtp_send_helpers.py [hunk 5]
```diff
@@ -151,3 +151,57 @@ def send_rtp_and_get_status_by_notice_number_via_rest(
...
+
+
+def send_rtp_v2_and_get_status_via_rest(
+    debtor_token: str,
+    creditor_token: str,
+    reader_token: str,
+    payer_id: str,
+    expected_send_status: int = 201,
+) -> str:
+    """Activate a debtor, send an RTP via REST API v2 (/rtps with Version: v2), and return the resulting RTP status."""
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply utils/rtp_send_helpers.py]: solved

File: utils/rtp_send_helpers.py [hunk 6]
```diff
@@ -151,3 +151,57 @@ def send_rtp_and_get_status_by_notice_number_via_rest(
...
+        debtor_token,
+        rtp_data["payer"]["payerId"],
+        secrets.debtor_service_provider.service_provider_id,
+    )
+    assert activation_response.status_code in (201, 409), "Error activating debtor"
+
+    send_response = send_rtp_v2(access_token=creditor_token, rtp_payload=rtp_data)
+    assert send_response.status_code == expected_send_status
+
+    resource_id = send_response.headers["Location"].split("/")[-1]
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply utils/rtp_send_helpers.py]: solved

File: utils/rtp_send_helpers.py [hunk 7]
```diff
@@ -151,3 +151,57 @@ def send_rtp_and_get_status_by_notice_number_via_rest(
...
+
+    activation_response = activate(
+        debtor_token,
+        rtp_data["payer"]["payerId"],
+        secrets.debtor_service_provider.service_provider_id,
+    )
+    assert activation_response.status_code in (201, 409), "Error activating debtor"
+
+    send_response = send_rtp_v2(access_token=creditor_token, rtp_payload=rtp_data)
+    assert send_response.status_code == expected_send_status
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply utils/rtp_send_helpers.py]: solved

File: utils/rtp_send_helpers.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (michele-fucito) [reply utils/rtp_send_helpers.py]: solved

File: utils/rtp_send_helpers.py [hunk 8]
```diff
@@ -151,3 +151,57 @@ def send_rtp_and_get_status_by_notice_number_via_rest(
...
+    return get_response.json()["status"]
+
+
+def send_rtp_v2_and_get_status_by_notice_number_via_rest(
+    debtor_token: str,
+    creditor_token: str,
+    reader_token: str,
+    payer_id: str,
+    expected_send_status: int = 422,
+) -> str:
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply utils/rtp_send_helpers.py]: solved

File: functional-tests/tests/send_rtp/test_RTP_send_MOCK.py [hunk 9]
```diff
@@ -392,3 +394,182 @@ def test_send_rtp_sync_rejected_no_links_THROUGH_WEB_API(
         secrets.mock_rjct_no_links_fiscal_code,
     )
     assert status == "REJECTED"
+
+
+# ── THROUGH_WEB_API_V2 variants ──────────────────────────────────────────────
+# These replicate the _THROUGH_WEB_API tests using the v2 REST send endpoint
+# (same URL /rtps but with Version: v2 header).
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply functional-tests/tests/send_rtp/test_RTP_send_MOCK.py:401]: I don't think this would be a good change since this would impact the report naming convention.

File: functional-tests/tests/send_rtp/test_RTP_send_MOCK.py [hunk 10]
```diff
@@ -392,3 +394,182 @@ def test_send_rtp_sync_rejected_no_links_THROUGH_WEB_API(
...
+
+
+@allure.epic("RTP Send")
+@allure.feature("RTP Send")
+@allure.story("Service provider sends an RTP")
+@allure.title("An RTP is sent through API - through Web API V2")
+@allure.tag("functional", "happy_path", "rtp_send")
+@pytest.mark.send
+@pytest.mark.happy_path
+def test_send_rtp_api_THROUGH_WEB_API_V2(
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply functional-tests/tests/send_rtp/test_RTP_send_MOCK.py:411]: I don't think this would be a good change since this would impact the report naming convention.

File: api/RTP_send_api.py [hunk 11]
```diff
@@ -4,18 +4,30 @@
...
     return requests.post(
-        headers={"Authorization": f"{access_token}", "Version": "v1"},
+        headers={"Authorization": f"{access_token}", "Version": version},
         url=SEND_RTP_URL,
         json=rtp_payload,
         timeout=HTTP_TIMEOUT,
     )
 
 
+def send_rtp(access_token: str, rtp_payload: dict) -> requests.Response:
```
Comment (luigidemarco-ship-it) [inline api/RTP_send_api.py:16]: Can we rename this function into "send_rtp_v1" in order to be symmetric to send_rtp_v2 function?

**Review by davideragusi29 — DISMISSED** [review]

File: api/RTP_send_api.py (hunk 11 above)
Comment (michele-fucito) [reply api/RTP_send_api.py:16]: Renaming  send_rtp  to  send_rtp_v1  would be a breaking change to the existing public API contract of this module. Other consumers (e.g. BDD tests, future integrations) rely on the current name. Since the asymmetry is purely cosmetic and carries no functional risk, I prefer to keep send_rtp  as-is and avoid an unnecessary breaking rename.

File: utils/rtp_send_helpers.py [hunk 12]
```diff
@@ -147,7 +367,61 @@ def send_rtp_and_get_status_by_notice_number_via_rest(
     )
     assert activation_response.status_code in (201, 409), "Error activating debtor"
 
-    send_response = send_rtp(access_token=creditor_token, rtp_payload=rtp_data)
+    send_response = send_rtp_v1(access_token=creditor_token, rtp_payload=rtp_data)
+    assert send_response.status_code == expected_send_status
+
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply utils/rtp_send_helpers.py]: solved

File: utils/rtp_send_helpers.py [hunk 13]
```diff
@@ -147,7 +367,61 @@ def send_rtp_and_get_status_by_notice_number_via_rest(
...
+        rtp_data["payer"]["payerId"],
+        secrets.debtor_service_provider.service_provider_id,
+    )
+    assert activation_response.status_code in (201, 409), "Error activating debtor"
+
+    send_response = send_rtp_v2(access_token=creditor_token, rtp_payload=rtp_data)
+    assert send_response.status_code == expected_send_status
+
+    resource_id = send_response.headers["Location"].split("/")[-1]
+
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply utils/rtp_send_helpers.py]: solved

File: api/RTP_send_api.py [hunk 14]
```diff
@@ -4,18 +4,30 @@
...
     )
 
 
+def send_rtp_v1(access_token: str, rtp_payload: dict) -> requests.Response:
+    """API to post a rtp request (Version: v1).
+    :returns: the response of the call.
+    :rtype: requests.Response
+    """
+    return _post_rtp(access_token, rtp_payload, "v1")
+
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply api/RTP_send_api.py]: solved

File: utils/rtp_send_helpers.py [hunk 15]
```diff
@@ -151,3 +371,57 @@ def send_rtp_and_get_status_by_notice_number_via_rest(
...
+        debtor_token,
+        rtp_data["payer"]["payerId"],
+        secrets.debtor_service_provider.service_provider_id,
+    )
+    assert activation_response.status_code in (201, 409), "Error activating debtor"
+
+    send_response = send_rtp_v2(access_token=creditor_token, rtp_payload=rtp_data)
+    assert send_response.status_code == expected_send_status
+
+    resource_id = send_response.headers["Location"].split("/")[-1]
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply utils/rtp_send_helpers.py]: solved

File: utils/rtp_send_helpers.py [hunk 16]
```diff
@@ -147,7 +367,61 @@ def send_rtp_and_get_status_by_notice_number_via_rest(
...
+
+    activation_response = activate(
+        debtor_token,
+        rtp_data["payer"]["payerId"],
+        secrets.debtor_service_provider.service_provider_id,
+    )
+    assert activation_response.status_code in (201, 409), "Error activating debtor"
+
+    send_response = send_rtp_v2(access_token=creditor_token, rtp_payload=rtp_data)
     assert send_response.status_code == expected_send_status
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply utils/rtp_send_helpers.py]: solved

**Review by davideragusi29 — APPROVED** [review]
